### PR TITLE
Issue detect useless generator pattern

### PR DESCRIPTION
**Pull request by Karine-Bauch**
In progress

### CODE REVIEW

**Comment by Karine-Bauch** [comment]
Error in issue comprehension
I will do another pull request later

Thanks

File: src/flake8_comprehensions/__init__.py [hunk 1]
```diff
@@ -199,6 +200,21 @@ def run(self) -> Generator[tuple[int, int, str, type[Any]], None, None]:
                         type(self),
                     )
 
+                elif (
+                    node.func.id == "sorted"
+                    and num_positional_args > 0
+                    and (
+                        isinstance(node.args[0], ast.GeneratorExp)
+                        and isinstance(node.args[0].elt, ast.Name)
+                    )
```
Comment (dorianturba) [inline src/flake8_comprehensions/__init__.py:209]: `and (... and ...)` can be rewritten `and ... and ...`

File: src/flake8_comprehensions/__init__.py [hunk 2]
```diff
@@ -199,6 +200,21 @@ def run(self) -> Generator[tuple[int, int, str, type[Any]], None, None]:
                         type(self),
                     )
 
+                elif (
+                    node.func.id == "sorted"
```
Comment (dorianturba) [inline src/flake8_comprehensions/__init__.py:204]: From the original issue:
> the (a for a in b) generator construct doesn't seem particularly useful and just adds an unnecessary context switch of the generator.

I think we can check every generator, not only those inside sorted call.
This mean that the check would be only the `isinstance` of node as `ast.GeneratorExp` and `isinstance` of `node.elt` as `ast.Name`. What do you think?

File: src/flake8_comprehensions/__init__.py (hunk 1 above)
Comment (Karine-Bauch) [reply src/flake8_comprehensions/__init__.py:209]: Oh, OK.
I thinked, in this issue, I should verify if isinstance of ast.Name only if it's a GeneratorExp 
Thanks

Comment (dorianturba) [reply src/flake8_comprehensions/__init__.py:209]: the form `and ... and ...` already mean that. If the first condition is falsy, the second won't be checked.

File: src/flake8_comprehensions/__init__.py (hunk 2 above)
Comment (Karine-Bauch) [reply src/flake8_comprehensions/__init__.py:204]: You should be right.
We should verify if the Generator do something, in all cases, not only for sorted call.
I need to put this on my desktop : KISS 👍

File: src/flake8_comprehensions/__init__.py (hunk 1 above)
Comment (Karine-Bauch) [reply src/flake8_comprehensions/__init__.py:209]: Oh, but yes... 🤦‍♀️

**Comment by adamchainz** [comment]
Closing because it seems work has stalled. Feel free to reopen if you continue work.

**Pull request closed without merging**